api.go

```diff
@@ -13,13 +13,21 @@ package gorqlite
 
 */
 
-import "bytes"
-import "encoding/json"
-import "errors"
-import "fmt"
-import "io/ioutil"
-import "net/http"
-import "time"
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+func init() {
+	// allow using self signed certificates
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
This should be done, for example, as a method on the `Connection` object. It should not be done for everyone.

Then people who want this behaviour can call the method. Make sense?

Reply:
Great idea, in fact the init function was more a quick fix than an elegant solution. Meanwhile, I put a logger on gorqlite and I identified that error come from Open function.

```
	log.SetFlags(log.LstdFlags | log.Lshortfile)
	func Open(connURL string) (Connection, error) {
	...
		err = conn.updateClusterInfo()  // this line return that error

```
Open function create and return a Connection variable. So, setting InsecureSkipVerify should be done before that but conn variable is still private (perhaps to allow multiple connections from the same application). A global variable setted by Open function only for https connections works (like in the example bellow), but I can't figured out if can be set it with a method of Connection object, here maybe you have a better idea..

```
	var InsecureSkipVerify bool
	
	func Open(connURL string) (Connection, error) {
		...
		f conn.wantsHTTPS {
			// allow using self signed certificates only for https connections
			http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: InsecureSkipVerify}
		}
		...
		err = conn.updateClusterInfo()
	
	// client application code
	rq.InsecureSkipVerify = true
	if db, err = rq.Open(node); err != nil {
		log.Fatal(err)
		return
	}
```